query/src/sql/sql_parser.rs

```diff
@@ -854,23 +854,108 @@ impl<'a> DfParser<'a> {
+                        return parser_err!(format!(
+                            "Keyword can not be parameter, got: {}",
+                            value
+                        ));
+                    }
+                    if let Some(quote) = quote_style {
+                        return parser_err!(format!(
+                            "Quote is not allowed in parameters, remove: {}",
+                            quote
+                        ));
+                    }
+                    if params.contains(&value) {
+                        return parser_err!(format!(
+                            "Duplicate parameter is not allowed, keep only one: {}",
+                            &value
+                        ));
+                    }
+
+                    expect_next_param = false;
+                    params.push(value);
+                }
+                Token::Comma => {
+                    expect_next_param = true;
+                }
+                other => {
+                    return parser_err!(format!("Expect words or comma, but got: {}", other));
+                }
+            }
+        }
+
+        if !found_right_paren {
+            return parser_err!("Can not find complete parameters, `)` is missing");
+        }
+        if expect_next_param {
+            return parser_err!("Found a redundant `,` in the parameters");
+        }
+
+        Ok(params)
+    }
+
+    fn parse_udf_definition_expr(&mut self, until_token: Vec<&str>) -> Result<String, ParserError> {
+        // Match ->
+        self.parser.expect_token(&Token::Minus)?;
+        let next_token = self.parser.next_token_no_skip();
+        if next_token != Some(&Token::Gt) {
+            return parser_err!(format!("Expected >, found: {:#?}", next_token));
+        }
+
+        let definition = self.consume_token_until_or_end(until_token).join("");
+        if definition.is_empty() {
+            return parser_err!("UDF definition can not be empty");
+        }
+
+        Ok(definition)
+    }
+
     fn parse_create_udf(&mut self) -> Result<DfStatement, ParserError> {
         let if_not_exists =
             self.parser
                 .parse_keywords(&[Keyword::IF, Keyword::NOT, Keyword::EXISTS]);
+
         let udf_name = self.parser.parse_literal_string()?;
-        self.parser.expect_token(&Token::Eq)?;
-        // TODO verify the definition as a legal expr
-        let definition = self.parser.parse_literal_string()?;
-        let description = if self.consume_token("DESC") {
-            self.parser.expect_token(&Token::Eq)?;
+        let as_token = Token::make_keyword("AS");
+        self.parser.expect_token(&as_token)?;
+
+        let desc_token = "DESC";
+        let parameters = self.parse_udf_parameters()?;
+        let definition = self.parse_udf_definition_expr(vec![desc_token])?;
+
+        let description = if self.consume_token(desc_token) {
+            self.parser.expect_token(&as_token)?;
```

Comment:
how about keep using `desc = 'desc_msg' `, it's strange to keep two `as` in a single sql.

Reply:
Updated.